src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3391,8 +3391,16 @@ internal override SyntaxNode GlobalAliasedName(SyntaxNode name)
         public override SyntaxNode NameExpression(INamespaceOrTypeSymbol namespaceOrTypeSymbol)
             => namespaceOrTypeSymbol.GenerateNameSyntax();
 
-        public override SyntaxNode TypeExpression(ITypeSymbol typeSymbol)
-            => typeSymbol.GenerateTypeSyntax();
+        private protected override SyntaxNode TypeExpression(ITypeSymbol typeSymbol, RefKind refKind)
+        {
+            var type = typeSymbol.GenerateTypeSyntax();
+            return refKind switch
+            {
+                RefKind.Ref => SyntaxFactory.RefType(type),
+                RefKind.RefReadOnly => SyntaxFactory.RefType(SyntaxFactory.Token(SyntaxKind.RefKeyword), SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword), type),
+                _ => type,
```

Comment:
❔ Does this also need to handle parameters (where `out` and `in` would be modifiers)?

Reply:
Nope.  That is handled at hte ParmaeterDecl creation level itself :)